tests/System.Text.Primitives.Tests/EncodingPerfTestsCorefxlab.cs

```diff
@@ -0,0 +1,341 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text.Utf8;
+using Xunit;
+using Microsoft.Xunit.Performance;
+
+namespace System.Text.Primitives.Tests
+{
+    public partial class EncodingPerfComparisonTests
+    {
+        public const ushort Utf16HighSurrogateFirstCodePoint = 0xD800;
+        public const ushort Utf16HighSurrogateLastCodePoint = 0xDBFF;
+        public const ushort Utf16LowSurrogateFirstCodePoint = 0xDC00;
+        public const ushort Utf16LowSurrogateLastCodePoint = 0xDFFF;
+
+        public const uint LastValidCodePoint = 0x10FFFF;
+
+        public const byte Utf8OneByteLastCodePoint = 0x7F;
+        public const ushort Utf8TwoBytesLastCodePoint = 0x7FF;
+        public const ushort Utf8ThreeBytesLastCodePoint = 0xFFFF;
+
+        public const int DataLength = 999;
+
+        public const int RandomSeed = 42;
+
+        [Benchmark]
+        [InlineData(DataLength, 0x0, Utf8OneByteLastCodePoint)]
+        [InlineData(DataLength, Utf8OneByteLastCodePoint + 1, Utf8TwoBytesLastCodePoint)]
+        [InlineData(DataLength, Utf8TwoBytesLastCodePoint + 1, Utf8ThreeBytesLastCodePoint)]
+        [InlineData(DataLength, Utf16HighSurrogateFirstCodePoint, Utf16LowSurrogateLastCodePoint)]
+        [InlineData(DataLength, 0x0, Utf8ThreeBytesLastCodePoint)]
+        [InlineData(DataLength, 0, 0, SpecialTestCases.AlternatingASCIIAndNonASCII)]
+        [InlineData(DataLength, 0, 0, SpecialTestCases.MostlyASCIIAndSomeNonASCII)]
+        public void EncodeFromUtf8toUtf8Corefxlab(int length, int minCodePoint, int maxCodePoint, SpecialTestCases special = SpecialTestCases.None)
```

Comment:
Would it be better to use type names as opposed to "corfxlab"? e.g. EncodeFromUtf8ToUtf16TextEncoder and EncodeFromUtf8ToUtf16Encoding